### PR TITLE
Getting started guide

### PR DESCRIPTION
**Pull request by lewisl9029**
Demo: https://metabase-getting-started.herokuapp.com/reference/guide

TODOs:
- [x] Add basic UI components and styling for guide details view
- [x] Add basic UI components and styling for guide edit view
- [x] Tweak font sizes to better match other data-ref sections
- [x] Adjust title margins to emphasize association to section below
- [x] Wire up see all links to appropriate locations
- [x] Remove see all dashboard link until we have an all dashboards view
- [x] Remove metrics values for page load performance
- [x] Wire up add metric/segment/table buttons on edit view
- [x] Fetch guide data from backend
- [x] Update guide data on edit
- [x] Add advanced segment/table selector using component from query builder
- [x] Wire up remove metric/segment/table workflow when ready
- [x] Add important fields multi-select for metrics
- [x] Add important fields for metrics to metric details and edit views
- [x] Remove items already selected in other selects from options
- [x] Implement pre-edit view and transitions
- [x] Add alternate edit view for empty sections
- [x] Validate form info (disallow submitting completely empty guide, invalid email, etc)
- [x] Fix metrics/segments list no longer updating after creating new metric/segment if section already visited
- [x] Fix empty metrics and segments inputs when navigating away from edit view then back (should be fixed when we implement pre-edit views)
- [x] Hide expand/collapse arrow for segment selector as per #3229

### CODE REVIEW

**Comment by tlrobinson** [comment]
Looking pretty good. A couple other things I noticed:

* No empty state for metrics.
* Segment/table picker feels a little wonky. If there are no segments it immediately shows me the segments empty state but then if I hit back it doesn't show me the segments section any more, which feels a little weird. I think we shouldn't show the empty state by default, and maybe always show the Segments section even if there aren't any segments?
* Should probably hide the "X" for the last metric and segment/table sections, since they aren't clickable.

**Comment by tlrobinson** [comment]
Should the titles on the left be wrapping when there's plenty of space for them?

screenshot 2016-08-15 16 51 05

**Comment by mazameli** [comment]
Yeah, we should increase the max-width on those.

**Comment by lewisl9029** [comment]
> No empty state for metrics.
Segment/table picker feels a little wonky. If there are no segments it immediately shows me the segments empty state but then if I hit back it doesn't show me the segments section any more, which feels a little weird. I think we shouldn't show the empty state by default, and maybe always show the Segments section even if there aren't any segments?

@tlrobinson Could you clarify what you meant by "empty state" here? Sorry I'm drawing a blank as to what that could refer to.

> Should probably hide the "X" for the last metric and segment/table sections, since they aren't clickable.

The "X" button actually clears the input on the last item in a section, so it does have a use there. Although it might be a good idea to hide it when the inputs are already empty. Any thoughts @mazameli?

Agreed on the title wrap. I'll try experimenting with some larger max-widths.

**Comment by tlrobinson** [comment]
@lewisl9029 screenshot 2016-08-15 17 04 58

Yeah, we shouldn't have buttons that don't do anything when clicked.

**Comment by tlrobinson** [comment]
Regarding the empty state, we should probably have a default empty state for `<Select />` with no options.

**Comment by lewisl9029** [comment]
Ah ok. Yeah I've noticed that too. The reason I haven't spent any time on that yet is because it should become irrelevant once I finish implementing the real empty states for getting started guide edit view: https://projects.invisionapp.com/share/VP7T9JI47#/screens/178416405

Agreed that Selects should handle empty options more gracefully though. Maybe we should extend it to take a `emptyPlaceholder` prop and disable the popover and display `emptyPlaceholder` when options are empty?

**Comment by mazameli** [comment]
Yeah, throughout the app, it would make sense to have dropdowns with no options in them be disabled.

> The "X" button actually clears the input on the last item in a section, so it does have a use there. Although it might be a good idea to hide it when the inputs are already empty. Any thoughts @mazameli?


Yeah, let's hide it when all of the inputs are empty. 👍

**Comment by camsaul** [comment]
omg 90 commits this is too much you guys should squash them 💔

**Comment by kdoh** [comment]
@tlrobinson When you have some time can you take a look at this from a code standpoint?

File: frontend/src/metabase/css/core/text.css [hunk 1]
```diff
@@ -111,3 +111,9 @@
     line-height: 1.4em;
     white-space: pre;
 }
+
+.text-measure { max-width: 620px; }
+
+.text-dark { color: #333333; }
```
Comment (tlrobinson) [inline frontend/src/metabase/css/core/text.css]: Is this close to another `text-*` we already have?

Comment (tlrobinson) [inline frontend/src/metabase/css/core/text.css]: Actually there already is a `text-dark` that's `--dark-color: #4C545B;` :(

Comment (tlrobinson) [inline frontend/src/metabase/css/core/text.css]: Looks like the #333333 one was being overridden by the real one anyway, I'm going to remove this.

**Comment by tlrobinson** [comment]
This is a big PR so I mostly focused on the modifications to existing parts of the app, which look fine aside from a few CSS things I fixed.

**Pull request merged**